cmd/kube-rbac-proxy/app/sanitazion.go

```diff
@@ -0,0 +1,70 @@
+package app
+
+import (
+	"bytes"
+	"encoding/json"
+	v1 "k8s.io/api/authentication/v1"
+	"strings"
+)
+
+// SanitizingFilter implements the LogFilter interface from klog with custom functions to detect and mask tokens.
+type SanitizingFilter struct{}
+
+// Filter is the filter function for non-formatting logging functions of klog.
+func (sf *SanitizingFilter) Filter(args []interface{}) []interface{} {
+	for i, v := range args {
+		if strValue, ok := v.(string); ok {
+			if containsTokenReview(strValue) {
+				args[i] = maskTokenInLog(strValue)
+			}
+		}
+	}
+	return args
+}
+
+// FilterF is the filter function for formatting logging functions of klog.
+func (sf *SanitizingFilter) FilterF(format string, args []interface{}) (string, []interface{}) {
+	for i, v := range args {
+		if strValue, ok := v.(string); ok {
+			if containsTokenReview(strValue) {
+				args[i] = maskTokenInLog(strValue)
+			}
+		}
+	}
+	return format, args
+}
+
+// FilterS is the filter function for structured logging functions of klog.
+func (sf *SanitizingFilter) FilterS(msg string, keysAndValues []interface{}) (string, []interface{}) {
+	for i, v := range keysAndValues {
+		if strValue, ok := v.(string); ok {
+			if containsTokenReview(strValue) {
+				keysAndValues[i] = maskTokenInLog(strValue)
+			}
+		}
+	}
+	return msg, keysAndValues
+}
+
+func containsTokenReview(logStr string) bool {
+	return strings.Contains(logStr, `"kind":"TokenReview"`)
+}
```

Comment:
Nit: this is quite shallow. It would make more sense to the code as is into the if clause.

(nit isn't a change required)